sashimi-webapp/src/logic/filemanager/file.js

```diff
@@ -0,0 +1,133 @@
+import storage from '../../database/storage';
+import Folder from './folder';
+
+/**
+* File Object
+*
+* @param {Number} fileID
+* @param {String} fileName
+* @param {String} filePath
+*/
+
+
+export default function File(fileID, fileName, filePath, parentFolder) {
+  this.id = fileID;
+  this.name = fileName;
+  this.path = filePath;
+  this.parentFolder = parentFolder;
+  console.log('new File created');
+}
+
+/**
+ * Remove file from the database
+ *
+ * @param {}
+ * @return {}
+ */
+File.prototype.remove = function remove() {
+  console.log('file.remove');
+  // Case 1a: file exist, file removed
+  // Case 1b: file exist, file not removed (error)
+  // Case 2: file does not exist (nothing removed)
+
+  return storage.deleteFile(this.path)
+    .then(() => {
+      const parentFolder = this.parentFolder;
+      const index = parentFolder.childFileList.findIndex(childFile => childFile.id === this.id);
+      parentFolder.childFileList.splice(index, 1);
+    })
+    .catch((error) => {
+      throw error;
+    });
+};
+
+/**
+ * Save data into file
+ *
+ * @param {String} data
+ * @return {}
+ */
+File.prototype.save = function save(data) {
+  console.log('file.save');
+  // Case 1a: file exist, file saved
+  // Case 1b: file exist, file not saved (error)
+  // Case 2: file does not exist (nothing saved)
+
+  return storage.saveFile(this.id, data);
```

Comment:
should use
```
storage.saveFile(this.id, data)
.then(() => true)
.catch(err => false);
```
something along this line would be better >.< at least catching the error.